sync/src/tests/helpers.rs

```diff
@@ -126,23 +132,63 @@ impl<'p, C> SyncIo for TestIo<'p, C> where C: FlushingBlockChainClient, C: 'p {
 		self.snapshot_service
 	}
 
+	fn private_transactions_provider(&self) -> &PrivateTransactionProvider {
+		&*self.private_provider
+	}
+
 	fn peer_session_info(&self, _peer_id: PeerId) -> Option<SessionInfo> {
 		None
 	}
 
 	fn eth_protocol_version(&self, _peer: PeerId) -> u8 {
-		63
+		ETH_PROTOCOL_VERSION_63
 	}
 
 	fn protocol_version(&self, protocol: &ProtocolId, peer_id: PeerId) -> u8 {
-		if protocol == &WARP_SYNC_PROTOCOL_ID { 2 } else { self.eth_protocol_version(peer_id) }
+		if protocol == &WARP_SYNC_PROTOCOL_ID { PAR_PROTOCOL_VERSION_3 } else { self.eth_protocol_version(peer_id) }
 	}
 
 	fn chain_overlay(&self) -> &RwLock<HashMap<BlockNumber, Bytes>> {
 		&self.overlay
 	}
 }
 
+pub fn push_block_with_transactions(client: &Arc<EthcoreClient>, engine: &EthEngine, transactions: &[SignedTransaction]) {
+	let block_number = client.chain_info().best_block_number as u64 + 1;
+
+	let mut b = client.prepare_open_block(Address::default(), (0.into(), 5000000.into()), Bytes::new());
+	b.set_difficulty(U256::from(0x20000));
+	b.set_timestamp(block_number * 10);
+
+	for t in transactions {
+		b.push_transaction(t.clone(), None).unwrap();
+	}
+	let b = b.close_and_lock().seal(engine, vec![]).unwrap();
+
+	if let Err(e) = client.import_block(b.rlp_bytes()) {
+		panic!("error importing block which is valid by definition: {:?}", e);
+	}
+
+	client.flush_queue();
+	client.import_verified_blocks();
+}
+
+/// Mock for emulution of async run of new blocks
+pub struct NewBlockMessage {
```

Comment:
Why public? It's not possible to access any fields nor to create such structure outside of this module.